demo/examples/generate_bind9_multipool_pools_yaml/generate_bind9_multipool_pools_yaml.go

```diff
@@ -0,0 +1,207 @@
+/*
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package main
+
+import (
+	"os"
+	"text/template"
+)
+
+type Pool struct {
+	Name        string
+	Description string
+	Attributes  map[string]string
+	NSRecords   []NSRecord
+	Nameservers []Nameserver
+	Targets     []Target
+	CatalogZone *CatalogZone // it is a pointer because it is optional
+}
+
+type NSRecord struct {
+	Hostname string
+	Priority int
+}
+
+type Nameserver struct {
+	Host string
+	Port int
+}
+
+type Target struct {
+	Type        string
+	Description string
+	Masters     []Master
+	Options     Options
+}
+
+type Master struct {
+	Host string
+	Port int
+}
+
+type Options struct {
+	Host           string
+	Port           int
+	RNDCHost       string
+	RNDCPort       int
+	RNDCConfigFile string
+}
+
+type CatalogZone struct {
+	FQDN    string
+	Refresh string
+}
+
+func main() {
+	attributes := map[string]string{
+		"type": "internal",
+	}
+
+	// Catalog zone is an optional section
+	catalogZone := &CatalogZone{
+		FQDN:    "example.org.",
+		Refresh: "60",
+	}
+
+	pool1 := Pool{
+		Name:        "default",
+		Description: "DevStack BIND Pool",
+		Attributes:  attributes,
+		NSRecords: []NSRecord{
+			{Hostname: "ns1.devstack.org.", Priority: 1},
+		},
+		Nameservers: []Nameserver{
+			{Host: "192.168.124.114", Port: 53},
+		},
+		Targets: []Target{
+			{
+				Type:        "bind9",
+				Description: "BIND Instance",
+				Masters: []Master{
+					{Host: "192.168.124.114", Port: 5354},
+				},
+				Options: Options{
+					Host:           "192.168.124.114",
+					Port:           53,
+					RNDCHost:       "192.168.124.114",
+					RNDCPort:       953,
+					RNDCConfigFile: "/etc/named/rndc.conf",
+				},
+			},
+		},
+		CatalogZone: catalogZone,
+	}
+	// Second pool definition
+	attributes = map[string]string{
+		"type": "external",
+	}
+	pool2 := Pool{
```

Comment:
This sample data should include multiple entries for binds and mdns servers.

Reply:
Done